### ISSUE TITLE
Inconsistent behaviour with asterisk-wildcard in array path

### ISSUE THREAD

**Issue by TobiasNx** [metafacture/metafacture-fix#121]
Since we do not have a ticket for that here. The asterisk wildcard in array paths seems to work sometimes, sometimes not and sometimes it breaks. I have set up a test with multiple usages of the `*`-wildcard and separated between working, breaking and no-doers. I am not sure if this is due to inconsistent array handling or due to the function. But I think it is good to collect them here:

https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/a8f314ce0c416073873553166eb9116e41a95cb7/data/testing/asteriskArray

Following functions work, also with arrays in arrays:
`prepend`
`append`
`replace`

`copy_field` works with single asterisk but not with arrays in arrays.

Following do nothing:
`split_field`
`sort_field`
`add_field`

Following break with asterisk:
`reverse`
`sum`
`rename`

**Comment by blackwinter** [comment]
There's a pattern w.r.t. how they're implemented internally:

- `append`, `prepend`, `replace_all`: `Record.transformFields()`
- `rename`, `reverse`, `sort_field`, `split_field`, `sum`: `Record.transformField()` [no `s`]
- `add_field`: `Record.append()`
- `copy_field`: `Record.copy()`

**Comment by blackwinter** [comment]
> `copy_field` works with single asterisk but not with arrays in arrays.

You mean it should work without `set_array(...)`? Otherwise, I don't see a difference between

```
set_array("TEST_TWO[]")
copy_field("test[].*", "TEST_TWO[].$append")
```

and 

```
set_array("TEST_3[]") # sic!
copy_field("test[].*", "TEST_THREE[].$append")
```

**Comment by blackwinter** [comment]
Oh, I guess you meant

```
set_array("TEST_4[]")
copy_field("nestedTest[].*.test[].*", "TEST_4[].$append")
```

So this is the _expected_ output for `TEST_THREE` then?

```json
"TEST_THREE" : [ {
  "1" : "One",
  "2" : "Two",
  "3" : "Three"
} ]
```

**Comment by TobiasNx** [comment]
No actually this is an error in my fix. I corrected it TEST_3 is connected to the reverse handling that is commented out:
https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/764464e662ad0cae2b176e3b6087784709dde8b9/data/testing/asteriskArray/asteriskArray.fix#L27-L28

**Comment by TobiasNx** [comment]
It should be: https://github.com/TobiasNx/fix-FunctionalReview-Testing/blob/67a97dce90a58890506ef25d86fa86ea9bba4501/data/testing/asteriskArray/expected.json#L48-L49

**Comment by blackwinter** [comment]
Okay. But `Test_4` is a typo again, it should be `TEST_4`.

**Comment by TobiasNx** [comment]
fixed (why are my code snippets not working?)

**Comment by blackwinter** [comment]
> why are my code snippets not working?

Which code snippets?

**Comment by blackwinter** [comment]
Turned all examples into unit tests.

**Comment by TobiasNx** [comment]
@blackwinter should we split this issue in single tickets?

**Comment by blackwinter** [comment]
No. Only if/when it turns out that the root causes are actually different and not fixable in one go.
